homeassistant/core.py

```diff
@@ -330,11 +330,49 @@ def _async_stop_handler(self, *args):
         self.exit_code = 0
         self.loop.create_task(self.async_stop())
 
+    @asyncio.coroutine
+    def _notify(self, message):
+        data = {'message': message,
+                'title': 'Restarting',
+                'notification_id': '{}.restart'.format(DOMAIN)}
+        yield from self.services.async_call(
+            'persistent_notification', 'create', data)
+
+    @asyncio.coroutine
+    def _async_check_config_and_restart(self):
+        """Restart Home Assistant if config is valid.
+
+        This method is a coroutine.
+        """
+        proc = yield from asyncio.create_subprocess_exec(
+            sys.argv[0],
+            '--script',
+            'check_config',
+            stdout=asyncio.subprocess.PIPE)
+        # Wait for the subprocess exit
+        (stdout_data, dummy) = yield from proc.communicate()
+        result = yield from proc.wait()
+        if result:
+            _LOGGER.error("check_config failed. Not restarting.")
+            content = str(stdout_data, 'utf-8')
+            # Print content to stdout so that it will have the colored
+            # pretty-printed error.
+            print(content)
+            # Put error cleaned from color codes in the error log so it
+            # will be visible at the UI.
+            _LOGGER.error(re.sub(r'\033\[[^m]*m', '', content))
+            yield from self._notify('Restart failed')
```

Comment:
Please remove this function and inline the calls inside this method. Also use `notification_id` so that you will overwrite previous restart hass notifications instead of crowding the UI.